assets/javascripts/discourse-assign/initializers/extend-for-assigns.js.es6

```diff
@@ -506,6 +506,24 @@ function initialize(api) {
     },
   });
 
+  api.modifyClass("component:topic-notifications-button", {
+    @discourseComputed(
+      "topic",
+      "topic.details.{notification_level,notifications_reason_id}"
+    )
+    notificationReasonText(topic) {
```

Comment:
Usually we try not override functions since any changes in core might inadvertently break this override. What we usually do is to extend core by providing an API for us to be able to register a callback in order to achieve the results we want.

Reply:
I'm not sure how I feel about this.

> override functions since any changes in core

I would think the API should not include `modifyClass` if the intention is for API users not to override functions.

> extend core by providing an API

wouldn't it be a bit of an overkill to have to create a new API each time we want to make a modification from a plugin? I understand overriding it this way would result in some coupling, but extending core each time we should not override a function feels like not a good long term solution. 🤔

Reply:
Another thought would be that plugins should have some form of compatibility for versions of Discourse. Having version control feels like a better long term solution, where plugin authors can freely use available APIs and be sure that it works with their current version of the app.

Them having a dependency on the core API change also doesn't seem right. It's a subjective problem.

Reply:
We discussed this privately and we feel like adding an extension to core for this small feature isn't worth it as this point. Since it isn't the end of the world if the description goes missing if the function override stops working, we're OK with this overriding the function here.